socks/socks.go

```diff
@@ -162,30 +159,33 @@ func ParseAddr(s string) Addr {
 }
 
 // Handshake fast-tracks SOCKS initialization to get target address to connect.
+// Read RFC 1928 for request and reply structure and sizes.
 func Handshake(rw io.ReadWriter) (Addr, error) {
-	// Read RFC 1928 section 4 for request and reply structure and sizes
-	buf := make([]byte, MaxReqLen)
-
-	_, err := rw.Read(buf) // SOCKS version and auth methods
-	if err != nil {
+	buf := make([]byte, 0xff)
```

Comment:
The magic number `0xff` should be defined a meaning constant. Reuse `MaxReqLen` should not hurt (remember the min 4KB page size).